src/functions/questionnaire/update-questionnaire/handler.ts

```diff
@@ -0,0 +1,49 @@
+import { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { questionnaireService } from "src/database/services";
+import QuestionnaireModel from "src/database/schemas/questionnaire/questionnaire";
+
+/**
+ * Lambda function to update a questionnaire
+ *
+ * @param event event
+ */
+const updateQuestionnaire: ValidatedEventAPIGatewayProxyEvent<typeof QuestionnaireModel> = async event => {
+  const { pathParameters, body, } = event;
+  const id = pathParameters?.id;
+
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: "Bad request"
+    };
+  }
+
+  const existingQuestionnaire = await questionnaireService.findQuestionnaire(id);
+  if (!existingQuestionnaire) {
+    return {
+      statusCode: 404,
+      body: `Questionnaire ${id} not found`
+    };
+  }
+
+  const questionnaireUpdates: QuestionnaireModel = {
```

Comment:
Make Import already. The model is now in develop.

```suggestion
  import type QuestionnaireModel from "src/database/models/questionnaire";
```